### DISCUSSION TITLE
Issue with HAFS Workflow Stopping after atm_prep on NCAR Derecho Machine

### DISCUSSION THREAD

**Discussion by muradme** [hafs-community/HAFS#282] [Q&A] [answered]
Hi everyone,

I am PhD student from University of Houston, and currently working on HAFS. I have successfully built and installed HAFS in NCAR derecho machine. But while running HAFS_rt_regional_atm, the workflow stops after running atm_prep. I have attached the screenshot of hafs_rt_status here, where it shows atm_prep is successful. but from hafs_atm_prep.log, I can see some following errors. I have also attached the log file here. I would greatly appreciate any insights or suggestions on how to resolve this issue. Thank you in advance for your help!

Best regards,
Murad

Screenshot: 
![Screenshot from 2024-07-30 13-05-22](https://github.com/user-attachments/assets/e23e70ea-5c85-4281-9331-8dadc86373d9)

log_file: 
[hafs_atm_prep.log](https://github.com/user-attachments/files/16433541/hafs_atm_prep.log)

**Comment by BijuThomas-NOAA** [comment, marked as answer]
It looks like prod_util is not loaded. Could you check if you have prod_util installed on Derecho and loaded for your job?

**Comment by muradme** [comment]
Hi, thanks for your suggestion.

I couldn't find prod_util installed on the NCAR Derecho system, but I found and installed NCEPLIBS-prod_utils from GitHub. Now, atm_prep runs without any error. However, the Rocoto workflow still does not start the next step, "ATM_IC." I couldn't find any log file to check its status.

Does this step need any additional input files to run, and what are these? If not, can you suggest any reasons why the "ATM_IC" task in the Rocoto workflow might not start and where I should look to solve this? I am attaching the workflow XML file, hafs_atm_prep.log file, and cronjob script here.

Best regards,
Murad

Attached files:
------------------
[hafs_atm_prep.log](https://github.com/user-attachments/files/16465644/hafs_atm_prep.log)
[cronjob_test2.sh.txt](https://github.com/user-attachments/files/16465657/cronjob_test2.sh.txt)
[hafs-HAFS_rt_regional_atm-13L-2020082512.xml.txt](https://github.com/user-attachments/files/16465663/hafs-HAFS_rt_regional_atm-13L-2020082512.xml.txt)

**Comment by BijuThomas-NOAA** [comment]
I don't have Derecho access. Run "rocotocheck" to see why the workflow is not moving.

**Comment by muradme** [comment]
Hi,

Thank you for pointing out the "rocotocheck" option. I am new to using the Rocoto workflow, and your guidance has been incredibly helpful.

I've attached the output from the rocotocheck command below. It seems that the file /glade/derecho/scratch/mkhondaker/project/momen/murad/project3/noscrub/hafs-input/COMGFSv16/gfs.20200825/12/atmos/gfs.t12z.atmanl.nc does not exist. I attempted to download it, but I was only able to access the last 10 days of GFS files. Could you suggest where I might find these files for previous years (i.e., 2019-2023)?

Thanks again for your assistance.

Best regards,
Murad

-----------------------------
rocotocheck -v -w hafs-HAFS_rt_regional_atm-13L-2020082512.xml -d hafs-HAFS_rt_regional_atm-13L-2020082512.db -c all -t atm_ic                                                                                                                                                                   
                                                                                                                                        
Task: atm_ic 
  account: UHOU0002
  command: /glade/work/mkhondaker/project/momen/murad/project3/jun_27//HAFS/jobs/JHAFS_ATM_IC                                                                                                 
  cores: 256   
  final: false
  jobname: hafs_atm_ic_13L_2020082512
  join: /glade/derecho/scratch/mkhondaker/HAFS_rt_regional_atm/2020082512/13L/hafs_atm_ic.log                                                                                                 
  maxtries: 1
  name: atm_ic                                                                                                                          
  nodes: 2:ppn=32:tpp=4                                                                                                                 
  nodesize: 128                                                                                                                         
  queue: main                                                                                                                           
  throttle: 9999999
  walltime: 00:15:00
  environment        
    COMhafs ==> /glade/derecho/scratch/mkhondaker/HAFS_rt_regional_atm/com/2020082512/13L                                                                                                     
    CONFhafs ==> /glade/derecho/scratch/mkhondaker/HAFS_rt_regional_atm/com/2020082512/13L/storm1.conf
    HOLDVARS ==> /glade/derecho/scratch/mkhondaker/HAFS_rt_regional_atm/com/2020082512/13L/storm1.holdvars.txt                                                                                
    HOMEhafs ==> /glade/work/mkhondaker/project/momen/murad/project3/jun_27//HAFS                                                                                                             
    NCTSK ==> 60       
    NET ==> hafs                                                                                                                        
    OMP_THREADS ==> 4     
    PDY ==> 20200825                                                                                                                    
    PYTHONPATH ==> /glade/work/mkhondaker/project/momen/murad/project3/jun_27//HAFS/ush                                                                                                       
    RUN ==> hfsa  
    RUN_ENVIR ==> dev  
    TOTAL_TASKS ==> 180                                                                                                                 
    USHhafs ==> /glade/work/mkhondaker/project/momen/murad/project3/jun_27//HAFS/ush                                                                                                          
    WHERE_AM_I ==> derecho
    WORKhafs ==> /glade/derecho/scratch/mkhondaker/HAFS_rt_regional_atm/2020082512/13L                                                                                                        
    cyc ==> 12          
    envir ==> test                                                             
    hafs_ver ==> v1.0.0                                                        
    jlogfile ==> /glade/derecho/scratch/mkhondaker/HAFS_rt_regional_atm/log/jlogfile                                                                                                          
  dependencies                                                                 
    AND is not satisfied                                                       
      atm_prep of cycle 202008251200 is SUCCEEDED                                                                                                                                                                                                                                                                            
      'NO'=='NO' is true                                                                       
      OR is satisfied                                                          
        input of cycle 202008251200 is not SUCCEEDED                           
        'NO'!='YES' is true                                                    
      Walltime is > 2020-08-25 15:25:00 UTC                                    
      AND is not satisfied                                                     
        /glade/derecho/scratch/mkhondaker/project/momen/murad/project3/noscrub/hafs-input/COMGFSv16/gfs.20200825/12/atmos/gfs.t12z.atmanl.nc does not exist                                                                                                                                                                                                                                  

Cycle: 202008251200                                                            
  Valid for this task: YES                                                                     
  State: active                                                                
  Activated: 2024-08-02 03:31:59 UTC                                           
  Completed: -                                                                                                                                                                                                                                                                                                               
  Expired: -                                                                                   

Job: This task has not been submitted for this cycle                                           

Task can not be submitted because:                                                             
  Dependencies are not satisfied

**Comment by BijuThomas-NOAA** [comment]
The 2023 data is archived on Jet/Orion/Hercules platforms. HPSS will have 2023 and prior to 2023 data

**Comment by muradme** [comment]
Hi, 
Thanks again for your help on this. Could you please provide me with the specific instructions or URL links or a link to the documentation on how to retrieve this data from HPSS? Additionally, I would appreciate it if you could share the directories or file paths for the 2023 hurricane input data available on the Jet platform (I am looking files for Hurricane Lee).
Best regards,
Murad

**Comment by BijuThomas-NOAA** [comment]
You can find GFS/GDAS/ENKF/RTOFS data to run HAFS here on Jet:

> /mnt/lfs4/HFIP/hwrf-data/hafs-input
> /mnt/lfs5/HFIP/hwrf-data/hafs-input

I am not aware of any documentation available on how to retrieve the needed data to run HAFS from HPSS.

**Comment by mkavulich** [comment]
Just chiming in on this discussion since there seems to be some confusion: the RDHPC HPSS referred to above is different from the NCAR HPSS. You can not access the hafs-input data directly from Derecho, you need an [RDHPCS](https://www.noaa.gov/organization/information-technology/getting-hpc-account) (NOAA HPC) account to access this data.

There is a long archive of GFS input data on AWS in GRIB2 format (https://registry.opendata.aws/noaa-gfs-bdp-pds/), but from what I can tell HAFS expects netcdf input files. @BijuThomas-NOAA Is there any way to run HAFS with GRIB2 data as input?

**Comment by ZhanZhang-NOAA** [comment]
To answer the questions above:
1. If DA is turned off (run_gsi=no), the AWS website includes all GFS files
required to run HAFS, including the GFS netcdf files, gfs.t??z.atmanl.nc
and gfs.t??z.sfcanl.nc for initial conditions;
2. If you change config. option from "ictype=gfsnetcdf" to
"ictype=gfsgrib2ab_0p25", you should be able to run HAFS with GRIB2 data as
input

**Comment by muradme** [comment]
Thank you all for your help with the input file sources. I successfully tested the AWS GFS files and was able to use them without any issues. For the ocean model forcing, I found a similar source for RTOFS files here: [RTOFS Files on AWS](https://noaa-nws-rtofs-pds.s3.amazonaws.com/index.html).

However, I wasn't able to find RTOFS data prior to 2024. Could you please advise on how I can access the RTOFS data for the years 2021-2023? Unfortunately, I don't have access to NOAA machines.

Thank you in advance!

**Reply by gspetro-NOAA** [reply]
Someone suggested that this site might have RTOFS data, but I'm not sure how much data is there or whether the site is still updated: https://polar.ncep.noaa.gov/global/data_access.shtml

**Comment by ZhanZhang-NOAA** [comment]
Yes, RTOFS started being uploaded to AWS in 2024. I don't know if there are
any other ways to get RTOFS data prior to 2024 unless you have access to
NOAA computer systems.

-Zhan

**Comment by MariaAristizabal-NOAA** [comment]
Hello Gillian,
Using the link you pasted, I got an ftp link (
ftp://ftpprd.ncep.noaa.gov/pub/data/nccf/com/rtofs/prod/) where you could
get some RTOFS data. Unfortunately I could not open it with my browser so I
can not tell you if it is updating or not. But in the past I used to get
the RTOFS output from there, but it only had about three days of the most
recent runs.
Best,
Maria

----
Maria Aristizabal Vargas, She/Her
Physical Scientist lll, Lynker at NOAA/EMC
NOAA Center for Weather & Climate Prediction
5830 University Research Court
College Park, MD 20740

**Comment by BinLiu-NOAA** [comment]
Currently, HAFS application does not support Derecho. With that, I am closing this discussion thread for now.
